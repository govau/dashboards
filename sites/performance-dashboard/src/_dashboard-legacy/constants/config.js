export const SCREEN_SM = 768;
export const DEFAULT_HERO_HEIGHT = 240;
