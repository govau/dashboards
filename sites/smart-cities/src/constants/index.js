export * from './misc';
export { default as INDICATORS } from './indicators';
export * from './categories';
export { default as CITIES } from './cities';
