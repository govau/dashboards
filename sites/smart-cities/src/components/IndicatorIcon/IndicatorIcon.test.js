import React from 'react';
import { shallow } from 'enzyme';
import IndicatorIcon from './IndicatorIcon';
import { STRINGS } from '../../constants';

jest.mock('../Portal/Portal', () => 'Portal');

const defaultProps = {
  contextual: true,
};

it('should match Snapshot', () => {
  const component = shallow(<IndicatorIcon {...defaultProps} />);

  expect(component.debug()).toMatchSnapshot();
});

it('should render a contextualIndicator Icon if the indicator is contextual', () => {
  const component = shallow(
    <IndicatorIcon
      {...defaultProps}
      contextual
    />
  );

  expect(component.find('Tooltip').prop('text')).toBe(STRINGS.CONTEXTUAL_DEFINITION);
  expect(component.find('Icon').prop('icon')).toBe('contextualIndicator');
});

it('should render a performanceIndicator Icon if the indicator is not contextual', () => {
  const component = shallow(
    <IndicatorIcon
      {...defaultProps}
      contextual={false}
    />
  );

  expect(component.find('Tooltip').prop('text')).toBe(STRINGS.PERFORMANCE_DEFINITION);
  expect(component.find('Icon').prop('icon')).toBe('performanceIndicator');
});
