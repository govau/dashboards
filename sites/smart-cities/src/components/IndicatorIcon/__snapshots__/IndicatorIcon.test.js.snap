// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<Tooltip borderColor={[undefined]} text=\\"Contextual indicators highlight the characteristics of a city that are not amenable to or potentially appropriate for local policy intervention but they can help to explain why a city performs the way it does.\\" className={[undefined]} style={{...}}>
  <Icon color={[undefined]} icon=\\"contextualIndicator\\" size={14} title=\\"\\" />
</Tooltip>"
`;
